### PR TITLE
Keep agent lists working when project records go stale

### PR DESCRIPTION
**Pull request by boudra**
### Linked issue

Refs #321

### Type of change

- [x] Bug fix
- [ ] New feature (with prior issue + design alignment)
- [ ] Refactor / code improvement
- [ ] Docs

### What does this PR do

Agent list RPCs no longer fail the whole request when one persisted agent points at a workspace whose parent project record is missing. Directory-style agent fetches skip unplaceable agents, while direct agent fetches still return the agent with `project: null`.

This keeps agent hydration usable when persisted workspace/project state is stale, without changing the protocol or the separate daemon-status RPC behavior.

### How did you verify it

Reproduced the failure first with a daemon e2e test that seeds stale persisted project/workspace state and calls the real websocket agent RPCs. Before the fix, the test failed with `DaemonRpcError: Project not found for workspace ... requestType=fetch_agents_request`; after the fix, it passes.

Commands run:

- `npx vitest run packages/server/src/server/daemon-e2e/agent-rpc-durability.e2e.test.ts --bail=1`
- `npx vitest run packages/server/src/server/session.workspaces.test.ts --bail=1`
- `npm run typecheck`
- `npm run lint`
- `npm run format`

Risk surface: persisted-state edge case only. Protocol shape is unchanged; the main behavior change is that unplaceable agents no longer poison the whole list response.

### Checklist

- [x] One focused change. Unrelated cleanups split out.
- [x] `npm run typecheck` passes
- [x] `npm run lint` passes
- [x] `npm run format` ran (Biome)
- [x] UI changes include screenshots or video for every affected platform (N/A: no UI changes)
- [x] Tests added or updated where it made sense

### CODE REVIEW

File: packages/server/src/server/daemon-e2e/agent-rpc-durability.e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      lastActivityAt: ORPHAN_ARCHIVED_AT,
+      lastUserMessageAt: null,
+      title: "Orphaned Archived Agent",
+      labels: {},
+      lastStatus: "closed",
+      lastModeId: "full-access",
+      config: null,
+      persistence: null,
+      archivedAt: ORPHAN_ARCHIVED_AT,
+    });
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/server/src/server/daemon-e2e/agent-rpc-durability.e2e.test.ts]: Fixed in 7411190af: moved the persisted-state setup into seedStaleAgentFixture so the test body focuses on the RPC calls and assertions.

File: packages/server/src/server/daemon-e2e/agent-rpc-durability.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    await client.connect();
+
+    const agents = await client.fetchAgents({
+      requestId: "req-agent-rpc-list",
+      filter: { includeArchived: true },
+    });
+    const orphanAgent = await client.fetchAgent({
+      requestId: "req-agent-rpc-detail",
+      agentId: orphanAgentId,
+    });
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/server/src/server/daemon-e2e/agent-rpc-durability.e2e.test.ts:43]: Fixed in 7411190af: the e2e now also calls fetch_agent_history_request against the stale project fixture and asserts the orphaned row is skipped.
